packages/teleport/src/Welcome/NewCredentials/NewPassword.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { Text, ButtonPrimary, ButtonText, Box } from 'design';
+import { Danger } from 'design/Alert';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import {
+  requiredPassword,
+  requiredConfirmedPassword,
+} from 'shared/components/Validation/rules';
+import { SliderProps } from 'teleport/components/StepSlider';
+import { Props as CredentialsProps, LoginFlow } from './NewCredentials';
+
+export default function NewPassword(props: Props) {
+  const {
+    submitAttempt,
+    resetToken,
+    isPasswordlessEnabled,
+    onSubmit,
+    auth2faType,
+    primaryAuthType,
+    password,
+    updatePassword,
+    switchFlow,
+    next,
+    refCallback,
+    willTransition,
+  } = props;
+  const [passwordConfirmed, setPasswordConfirmed] = useState('');
+  const mfaEnabled = auth2faType !== 'off';
+
+  function handleOnSubmit(
+    e: React.MouseEvent<HTMLButtonElement>,
+    validator: Validator
+  ) {
+    e.preventDefault(); // prevent form submit default
+
+    if (!validator.validate()) {
+      return;
+    }
+
+    if (mfaEnabled) {
+      next();
+      return;
+    }
+
+    onSubmit(password);
+  }
+
+  function switchToPasswordlessFlow(e, isBack = false) {
+    e.preventDefault();
+    switchFlow('passwordless', isBack);
```

Comment:
Hmm I have problems with figuring out what `isBack` means, can we find more meaningful name?